src/box/errcode.h

```diff
@@ -115,286 +115,286 @@ struct errcode_record {
+	_(ER_TUPLE_FORMAT_LIMIT, 16,		"Tuple format limit reached", "limit", UINT) \
+	_(ER_DROP_PRIMARY_KEY, 17,		"Can't drop primary key while secondary keys exist", "space", STRING) \
+	_(ER_KEY_PART_TYPE, 18,			"Supplied key field type does not match index part type", "fieldno", UINT, "part_type", STRING) \
+	_(ER_EXACT_MATCH, 19,			"Invalid key part count in an exact match", "expected", UINT, "actual", UINT) \
+	_(ER_INVALID_MSGPACK, 20,		"Invalid MsgPack", "issue", STRING) \
+	/* ER_PROC_RET, 21, Unused */ \
 	_(ER_TUPLE_NOT_ARRAY, 22,		"Tuple/Key must be MsgPack array") \
-	_(ER_FIELD_TYPE, 23,			"Tuple field %s type does not match one required by operation: expected %s, got %s") \
-	_(ER_INDEX_PART_TYPE_MISMATCH, 24,	"Field %s has type '%s' in one index, but type '%s' in another") \
-	_(ER_UPDATE_SPLICE, 25,			"SPLICE error on field %s: %s") \
-	_(ER_UPDATE_ARG_TYPE, 26,		"Argument type in operation '%c' on field %s does not match field type: expected %s") \
-	_(ER_FORMAT_MISMATCH_INDEX_PART, 27,	"Field %s has type '%s' in space format, but type '%s' in index definition") \
-	_(ER_UNKNOWN_UPDATE_OP, 28,		"Unknown UPDATE operation #%d: %s") \
-	_(ER_UPDATE_FIELD, 29,			"Field %s UPDATE error: %s") \
+	_(ER_FIELD_TYPE, 23,			"Tuple field type does not match one required by operation", "field", STRING, "expected", STRING, "actual", STRING) \
+	_(ER_INDEX_PART_TYPE_MISMATCH, 24,	"Field type mismatch in indexes", "field", STRING, "type_1", STRING, "type_2", STRING) \
+	_(ER_UPDATE_SPLICE, 25,			"SPLICE error", "field", STRING, "details", STRING) \
+	_(ER_UPDATE_ARG_TYPE, 26,		"Argument type does not match expected field type", "operation", CHAR, "field", STRING, "field_type", STRING) \
+	_(ER_FORMAT_MISMATCH_INDEX_PART, 27,	"Field type mismatch in space format and index definition", "field", STRING, "space_type", STRING, "index_type", STRING) \
+	_(ER_UNKNOWN_UPDATE_OP, 28,		"Unknown UPDATE operation", "operation_index", INT, "operation", STRING) \
+	_(ER_UPDATE_FIELD, 29,			"Fields UPDATE error", "field", STRING, "details", STRING) \
 	_(ER_FUNCTION_TX_ACTIVE, 30,		"Transaction is active at return from function") \
-	_(ER_KEY_PART_COUNT, 31,		"Invalid key part count (expected [0..%u], got %u)") \
-	_(ER_PROC_LUA, 32,			"%s") \
-	_(ER_NO_SUCH_PROC, 33,			"Procedure '%.*s' is not defined") \
-	_(ER_NO_SUCH_TRIGGER, 34,		"Trigger '%s' doesn't exist") \
-	_(ER_NO_SUCH_INDEX_ID, 35,		"No index #%u is defined in space '%s'") \
-	_(ER_NO_SUCH_SPACE, 36,			"Space '%s' does not exist") \
-	_(ER_NO_SUCH_FIELD_NO, 37,		"Field %d was not found in the tuple") \
-	_(ER_EXACT_FIELD_COUNT, 38,		"Tuple field count %u does not match space field count %u") \
-	_(ER_FIELD_MISSING, 39,			"Tuple field %s required by space format is missing") \
+	_(ER_KEY_PART_COUNT, 31,		"Invalid key part count", "maximum", UINT, "actual", UINT) \
+	_(ER_PROC_LUA, 32,			"Error in Lua code", "details", STRING) \
```

Comment:
"Error in Lua function" probably would be more consistent with the code name. Using "function" instead of "procedure" for consistency with the codes below.

Reply:
Done. 

```diff
+       _(ER_PROC_LUA, 32,                      "Error in Lua function", "details", STRING) \
```